src/views/webviewController.ts

```diff
@@ -105,6 +106,54 @@ export default class WebviewController {
     this._themeChangedSubscription?.dispose();
   }
 
+  handleWebviewOpenFileChooserAttempt = async ({
+    panel,
+    fileChooserOptions,
+    requestId,
+  }: {
+    panel: vscode.WebviewPanel;
+    fileChooserOptions: FileChooserOptions;
+    requestId: string;
+  }): Promise<void> => {
+    let files;
+
+    try {
+      files = await vscode.window[
+        fileChooserOptions.mode === 'open' ? 'showOpenDialog' : 'showSaveDialog'
+      ]({
+        defaultUri: vscode.Uri.from({
+          path: fileChooserOptions.electronFileDialogOptions?.defaultPath,
+          scheme: 'file',
+        }),
+        [`${fileChooserOptions.mode}Label`]:
+          fileChooserOptions.electronFileDialogOptions?.buttonLabel,
+        filters: fileChooserOptions.electronFileDialogOptions?.filters?.reduce(
+          (acc, filter) => {
+            acc[filter.name] = filter.extensions;
+            return acc;
+          },
+          {}
+        ),
+        title: fileChooserOptions.electronFileDialogOptions?.title,
+      });
+    } catch (error) {
+      void vscode.window.showErrorMessage(
+        `Unable to open file chooser dialog: ${error}`
+      );
+    }
+
+    void panel.webview.postMessage({
+      command: MESSAGE_TYPES.OPEN_FILE_CHOOSER_RESULT,
+      fileChooserResult: {
+        canceled: false,
```

Comment:
We never send canceled - is this correct? Should we not report when a user cancels the operation?

Reply:
The vscode electron dialog does not return a canceled state. We keep `canceled` here to satisfy the `createElectronFileInputBackend` API. We could assume that if returned files are undefined then the user has hanseled file selection, but from UX perspective it will be the same behaviour, since we won't show any toast if a user manually clicked the cancel button.